casper/src/test/scala/coop/rchain/casper/MultiParentCasperAddBlockSpec.scala

```diff
@@ -124,6 +124,17 @@ class MultiParentCasperAddBlockSpec extends FlatSpec with Matchers with Inspecto
     }
   }
 
+  it should "create valid blocks when peek syntax is present in a deploy" in effectTest {
+    HashSetCasperTestNode.standaloneEff(genesis).use { node =>
+      val source = " for(@x <<- @0){ Nil } | @0!(0) "
+      for {
+        deploy <- ConstructDeploy.sourceDeployNowF[Effect](source)
+        block  <- node.addBlock(deploy)
+        result <- node.casperEff.contains(block.blockHash) shouldBeF true
+      } yield result
```

Comment:
Is there code added in this PR that - when removed - breaks this test? If not, we need at least one more test.

Reply:
Yes there is, read the commit message

Reply:
Is it then right to say that peek - as we have it in dev - doesn't actually work?

Reply:
Yes, Casper is unable to correctly do the replay. It works in the rholang CLI though ;-)

Reply:
@tymm I think we've found the reason replay of registry lookups doesn't work,  please rebase on this PR to confirm.

Sorry for not catching that, should've demanded a more e2e-ish test during the original review.

Reply:
demanded? 🤔

Reply:
I meant requested, and not approved until the request is met. I guess that's a demand? :joy:

Reply:
One last question: do I understand right that rspace populates peeks in its event log, casper didn't have peeks in the proto, so replay would encounter a different COMM than in the provided (casper) event log (one with the peek indices set)?

Reply:
This is correct

Reply:
In the future, adding similar explanation to the commit message would be helpful :)